src/nvme/private.h

```diff
@@ -180,7 +180,8 @@ int json_dump_tree(nvme_root_t r);
 nvme_ctrl_t __nvme_lookup_ctrl(nvme_subsystem_t s, const char *transport,
 			       const char *traddr, const char *host_traddr,
 			       const char *host_iface, const char *trsvcid,
-			       nvme_ctrl_t p);
+			       nvme_ctrl_t p, const char *subsysnqn,
+			       bool dc_only);
```

Comment:
`dc_only` could be an enum type which makes it a bit simpler to read on the caller side IMO.

Reply:
That's an interesting idea. I didn't really like this parameter in the first place, but we're a bit stuck. The issue is that for the longest time DCs only had the well-known NQN (nqn.2014-08.org.nvmexpress.discovery). This meant that by looking at the subsysnqn we could easily determine whether we were dealing with an I/O or Discovery controller.

With TP8013, however, DCs can now have a unique NQN. On top of that, one can create a connection to a DC using the well-known NQN, but during the identify phase, the NQN reported through the sysfs can change to a unique NQN. So the config subsysnqn may not match the NQN reported by the sysfs. There are other attributes that can be read from the sysfs to make differentiate a discovery from an I/O controller.

Long story short, we could indeed use an enum here. I can make the change.

Reply:
Actually, this can be simplified. We don't need an additional parameter to identify the controller type. I went down this path because of how it was done in nvme-cli, but the reality is that we only need to use special handling when dealing with the well-known NQN (`"nqn.2014-08.org.nvmexpress.discovery"`). For all other cases, the unique `subsysnqn` is all we need to worry about. 

Let me clean this up and resubmit.